torch/distributed/device_mesh.py

```diff
@@ -794,12 +794,39 @@ def from_group(
         ) -> "DeviceMesh":
             """
             Constructs a :class:`DeviceMesh` with ``device_type`` from an
-            existing :class:`ProcessGroup`.
+            existing :class:`ProcessGroup` or a list of existing :class:`ProcessGroup`.
 
             The constructed device mesh has number of dimensions equal to the
-            number of groups passed. If more than one group is passed, then the
-            ``mesh`` argument is required.
+            number of groups passed. For example, if a single process group is passed in,
+            the resulted DeviceMesh is a 1D mesh. If a list of 2 process groups is passed in,
+            the resulted DeviceMesh is a 2D mesh.
+
+            If more than one group is passed, then the ``mesh`` and ``mesh_dim_names`` arguments
+            are required. The order of the process groups passed in determines the topology of
+            the mesh. For example, the first process group will be the 0th dimension of the DeviceMesh.
+            The `mesh` tensor passed in must have the same number of dimensions as the number of process
+            groups passed in, and the order of the dimensions in the `mesh` tensor must match the order
+            in the process groups passed in.
+
+            Args:
+                group (ProcessGroup or list[ProcessGroup]): the existing ProcessGroup
+                    or a list of existing ProcessGroups.
+                device_type (str): The device type of the mesh. Currently supports: "cpu",
+                    "cuda/cuda-like". Passing in a device type with a GPU index, such as "cuda:0",
+                    is not allowed.
+                mesh (torch.Tensor or ArrayLike, optional): A multi-dimensional array or an
+                    integer tensor describing the layout of devices, where the IDs are global IDs
+                    of the default process group. Default is None.
+                mesh_dim_names (tuple[str], optional): A tuple of mesh dimension names to assign
+                    to each dimension of the multi-dimensional array describing the layout of devices.
+                    Its length must match the length of `mesh_shape`. Each string in `mesh_dim_names`
```

Comment:
`mesh.shape`?

Reply:
`mesh` and `mesh_dim_names` are asymmetrical. One can be positional  and another one must be keyword, but they are used together.

Reply:
Ye...It is inconsistent. This is due to the fact that originally when we introduced `init_device_mesh`, `mesh_dim_names` is not required, and it is also an optional keyword arg...

https://github.com/pytorch/pytorch/blob/main/torch/distributed/device_mesh.py#L940-L945

Also, the mesh tensor should be mesh_shape in order to be consistent with `init_device_mesh`, so it may be good to do breaking change now instead of later.